File: launcher/driver/node/src/main/scala/dev/argon/driver/sjs/command/CommandFilePath.scala
package dev.argon.driver.sjs.command

import dev.argon.driver.scalaApi.command.CommandFilePathPlatformSpecific
import dev.argon.io.PathLike
import nobleidl.core.JSAdapter

type CommandFilePath = PathLike

object CommandFilePath {
  def jsAdapter(): JSAdapter[CommandFilePath, PathLike] =
    JSAdapter.identity
}

File: launcher/driver-api/jvm/src/main/java/dev/argon/driver/api/command/CommandFilePath.java
package dev.argon.driver.api.command;

import dev.argon.nobleidl.runtime.graaljsInterop.JSAdapter;
import dev.argon.nobleidl.runtime.graaljsInterop.JSExecutor;
import org.graalvm.polyglot.Context;
import org.graalvm.polyglot.Value;

import java.nio.file.Path;

public class CommandFilePath {
	private CommandFilePath() {}

	public static JSAdapter<Path> jsAdapter() {
		return new JSAdapter<>() {
			@Override
			public Path fromJS(Context context, JSExecutor jsExecutor, Value value) {
				return Path.of(value.asString());
			}

			@Override
			public Value toJS(Context context, JSExecutor jsExecutor, Path path) {
				return context.asValue(path.toString());
			}
		};
	}
}
